crypto/crypto.go

```diff
@@ -37,3 +37,22 @@ func GenerateOtp(digits int) (string, error) {
 	otp := fmt.Sprintf(expr, val.String())
 	return otp, nil
 }
+
+// GenerateOtpFromCharset generates a random n-length otp from a charset
+func GenerateOtpFromCharset(length int, charset string) (string, error) {
+	b := make([]byte, length)
+	for i := range b {
+		val, err := rand.Int(rand.Reader, big.NewInt(int64(len(charset))))
```

Comment:
is this how we did it previously? If not have we checked that it's a cryptographically secure way of doing random number generation in go?

Reply:
ah yeah looks ok: https://pkg.go.dev/crypto/rand